BringYourOwnBoard.md

```diff
@@ -0,0 +1,157 @@
+
+You'll also notice a section in here that's focused on providing a method for rolling back an update. That's why we have the firmware flag `have_updated` and `boot_attempts`. Once there has been an update we set the `have_updated` flag to `1` which indicates to go through the loop, but we only rollback when we've attempted to boot into the devices three times. That causes an auto fallback. 
+
+## recipes-core 
+
+### What does it do? 
+
+The recipes-core meta layer is responsible for finalizing the physical image itself. That is this layer edits the core image being generated and adds all relevant information, operating system setup, and adds the correct file system layout to allow Device Update for IoT Hub to perform it's updates. 
+
+There's only two things this section does to edit the base raspberrypi image to support Device Update. 
+
+First is to add a `fstab` file to the core image that mounts the required partitions by appending the base-files_ within the raspberrypi base image. We accomplish this by overriding the default [fstab file](https://git.yoctoproject.org/poky/tree/meta/recipes-core/base-files/base-files/fstab?h=kirkstone-4.0.17) provided by the Yocto Project and adding our [fstab file](./recipes-core/base-files/base-files/raspberrypi4-64/fstab). 
+
+Second the adu-base-image is defined by [adu-base-image.bb](./recipes-core/images/adu-base-image.bb) and the partitions on the image are created using our [`adu-raspberrypi.wks`](./wic/adu-raspberrypi.wks) script. You can read more about what's going on in that file below and checkout the [wic tool documentation here.](https://docs.yoctoproject.org/dev-manual/wic.html?highlight=wic#creating-partitioned-images-using-wic). A part of building the image is specifying the items that must be installed in this particular image. We use this to define which elements from [meta-azure-device-update](https://github.com/Azure/meta-azure-device-update.git) we're going to be using in this image. This is like a dependency list. It tells Yocto build all of this first and then run the `do_install()` sections of the individual items before you giving me my image. 
+
+### What do I need to think about? 
+
+Whatever system you're doing you need a way to make the A, B, and adu partitions. You can use `wic` like we do or if the tooling from your BSP doesn't support / won't work with `wic` they will provide another method for accomplishing the same task. 
+
+As for making your image you can peruse what we put in our image (look at the `IMAGE_INSTALL` section in the [adu-base-image file](./recipes-core/images/adu-base-image.bb) ), but the most important thing is to include the following assuming you are making no changes to `meta-azure-device-update`: 
```

Comment:
Punctuation nitpick: for "include the following assuming you are making no changes to", can we add a comma or parenthesis as follows:

"include the following, assuming you are..."
OR
"include the following (assuming you are...)"